src/seedu/addressbook/AddressBook.java

```diff
@@ -1209,4 +1222,18 @@ private static String removePrefixSign(String s, String sign) {
         return new ArrayList(Arrays.asList(toSplit.trim().split("\\s+")));
     }
 
+    /*
+     * ============================
+     *     COMPARATOR METHODS
+     * ============================
+     */
+    private static Comparator<String[]> comparatorForAlphabeting() {
+		return new Comparator<String[]>() {
+			@Override
+			 	public int compare(String[] o1, String[] o2) {
+			 		if (o1[0].toLowerCase().charAt(0) > o2[0].toLowerCase().charAt(0)) return 1;
```

Comment:
Comparing only the first character of the strings may give the wrong result. For example, if `o1[0] = "ab"` and `o2[0] = "aa"`, this function should return 1 as `"ab" > "aa"`, but it returns -1 instead! Also, if `o1[0] = "a"` and `o2[0] = "a"`, this function should return 0 as `"a" == "a"`, but this function returns -1 instead!

Also, `PERSON_DATA_INDEX_NAME` should probably be used instead of hard-coding `0`.